File: solutions/Python/1456.MaximumNumberOfVowelsInASubstringOfGivenLength.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels: set[str] = {"a", "e", "i", "o", "u"}
        current_vowels: int = sum(1 for char in s[:k] if char in vowels)
        max_vowels: int = current_vowels

        for i in range(k, len(s)):
            if s[i - k] in vowels:
                current_vowels -= 1
            if s[i] in vowels:
                current_vowels += 1
            max_vowels = max(max_vowels, current_vowels)

        return max_vowels

File: solutions/JavaScript/1456.MaximumNumberOfVowelsInASubstringOfGivenLength.js
var maxVowels = function (s, k) {
    const vowels = new Set(['a', 'e', 'i', 'o', 'u']);
    let currentVowels = 0;
    for (let i = 0; i < k; i++) {
        if (vowels.has(s[i])) {
            currentVowels++;
        }
    }
    let maxVowels = currentVowels;

    for (let i = k; i < s.length; i++) {
        if (vowels.has(s[i - k])) {
            currentVowels--;
        }
        if (vowels.has(s[i])) {
            currentVowels++;
        }
        maxVowels = Math.max(maxVowels, currentVowels);
    }

    return maxVowels;
};